### ISSUE TITLE
Linux app?

### ISSUE THREAD

**Issue by 9acca9** [libgenapps/LibgenDesktop#47]
HI... so, there is no linux app? i can get the books from libgen with another app?
Thanks!

**Comment by ThatOneCalculator** [comment]
I tried building from source to no avail.

```
LibgenDesktop on  master [?] via .NET 6.0.100   nuget restore
MSBuild auto-detection: using msbuild version '15.0' from '/usr/lib/mono/msbuild/15.0/bin'.
All packages listed in packages.config are already installed.

LibgenDesktop on  master [?] via .NET 6.0.100   msbuild .
Microsoft (R) Build Engine version 16.10.1 for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
Build started 1/3/2022 2:57:33 PM.
Project "/tmp/LibgenDesktop/LibgenDesktop.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Debug x64|Any CPU".
ValidateProjects:
  The project "LibgenDesktop.Setup" is not selected for building in solution configuration "Debug x64|Any CPU".
Project "/tmp/LibgenDesktop/LibgenDesktop.sln" (1) is building "/tmp/LibgenDesktop/LibgenDesktop/LibgenDesktop.csproj" (2) on node 1 (default targets).
/usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2218,5): error MSB3248: Parameter "AssemblyFiles" has invalid value "/usr/lib/mono/4.5-api/mscorlib.dll". Could not load file or assembly 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. [/tmp/LibgenDesktop/LibgenDesktop/LibgenDesktop.csproj]
Done Building Project "/tmp/LibgenDesktop/LibgenDesktop/LibgenDesktop.csproj" (default targets) -- FAILED.
Done Building Project "/tmp/LibgenDesktop/LibgenDesktop.sln" (default targets) -- FAILED.

Build FAILED.

"/tmp/LibgenDesktop/LibgenDesktop.sln" (default target) (1) ->
"/tmp/LibgenDesktop/LibgenDesktop/LibgenDesktop.csproj" (default target) (2) ->
(ResolveAssemblyReferences target) ->
  /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2218,5): error MSB3248: Parameter "AssemblyFiles" has invalid value "/usr/lib/mono/4.5-api/mscorlib.dll". Could not load file or assembly 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. [/tmp/LibgenDesktop/LibgenDesktop/LibgenDesktop.csproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.37
```

So for now I'm using the 64 bit MSI under wine.

**Comment by saltedcoffii** [comment]
@libgenapps What's the status on a Linux application?

**Comment by interdisziplina** [comment]
Same, I also want a Linux app

**Comment by interdisziplina** [comment]
If you are on Linux and don't mind using your terminal, I recommend [audstanley/libgen-tui](https://github.com/audstanley/libgen-tui).

**Comment by adrianlopezgalera** [comment]
There are things that I will never understand in this life. Among them is the fact that an organization against DRM and information control launches an exclusive application for a proprietary operating system, whose company promotes values contrary to those defended here. In your eagerness to reach people you only contribute to the dependence on technologies that exert injustices.

Regards.

**Comment by interdisziplina** [comment]
@Julvenzor 

I disagree here. The work the LibGen developer(s) have done is insane, and I would even consider it to be one of the best projects in humankind. To demand a slightly more comfortable (remember that you can always use LibGen from your browser) experience for one operating system is imo a very ungrateful and privileged statement. Please remember that open-source developers are humans too and usually do their work for free, so you should be thankful for their work and not demand even more.

I myself am a GNU/Linux user and would also like a GUI application for LibGen. But I would never criticize developers that made LibGen more, not less, available, for this application's unavailability on GNU/Linux.

**Comment by adrianlopezgalera** [comment]
> @Julvenzor
> 
> I disagree here. The work the LibGen developer(s) have done is insane, and I would even consider it to be one of the best projects in humankind. To demand a slightly more comfortable (remember that you can always use LibGen from your browser) experience for one operating system is imo a very ungrateful and privileged statement. Please remember that open-source developers are humans too and usually do their work for free, so you should be thankful for their work and not demand even more.
> 
> I myself am a GNU/Linux user and would also like a GUI application for LibGen. But I would never criticize developers that made LibGen more, not less, available, for this application's unavailability on GNU/Linux.

I have not issued any value judgment on anyone. You, on the other hand, are judging me and calling me "ungrateful." I have pointed out, with arguments, a logical contradiction, nothing more. Today there are many cross-platform languages that allow you to develop once and compile on other systems. There is no difficulty if there is a good prior approach.

Please, don't talk "privileges" and shit like that because you have no idea of my origin, of my life and of the difficulties I've experienced. You also don't know what I do in my day to day to fight for a better world. I am a vegan activist and I try to save animals while many individuals, who raise different flags, are dedicated to revindicating their personal oppressions while participating in the exploitation and systematic murder of millions of animals. All this, in my case and in that of others, is argumentatively irrelevant. I'll give you some advice: stop judging others and learn to bear criticism.

Regards.

**Comment by lakshaysethi** [comment]
Wine + msi = this app on linux , what is the issue?

**Comment by 8dcc** [comment]
>ungrateful and privileged statement

>Rust pfp with pride flag
>Rust
>any/all pronouns
>Pronouns

That's all, your honour.

**Comment by ThatOneCalculator** [comment]
Don't judge someone's thoughts, ideas, or opinions based on their identity.

**Comment by 8dcc** [comment]
str8 up not judging peace to every male and females on the eart

**Comment by ThatOneCalculator** [comment]
You know exactly what you're doing when you say "every male and female". If you seriously refuse to recognize and respect people who don't fall into either of those categories (either biologically or in gender expression), feel free to leave this issue and repository. And before you bring out the "biologically there are only two genders" argument, do your research on intersex people. They have been around for as long as everyone else.

**Comment by 8dcc** [comment]
finna fade cuz ma karmas gone bankrupt 😭

**Comment by BlackBallSack** [comment]
@ThatOneCalculator You don't seem to be very intelligent, since you don't know the difference between an illness and a gender, unless you are saying that all other genders(they don't exist) is mental illness. best regards non mongoloid.

**Comment by tigros** [comment]
ha How do you know if someone is a vegan: they WILL tell you! more proof of that right here.

**Comment by adrianlopezgalera** [comment]
> ha How do you know if someone is a vegan: they WILL tell you! more proof of that right here.

Not being vegan means participating in violence against animals. In the same way, I can tell who is a speciesist who enjoys the suffering of animals.

**Comment by eyalroz** [comment]
Let's try to be a bit more practical: I realize this application is written in C#, which is a Microsoft-associated and Windows-associated language AFAIK. But - I also know that there is some level of support for running C# on Linux. Microsoft claims it can be installed:

https://learn.microsoft.com/en-us/dotnet/core/install/linux

and maybe there are other alternatives. I've never used C# on Linux personally, so I'm just "spitballing" here. 

Another alternative is porting the application. cloc says there are ~27K lines of C# code. Since this is a client for a network service, not a super-hyper-duper-special C# project, I would assume a reasonable developer who knows both C# and a reasonably-multiplatform language could do it. Perhaps even the developers of the app itself could be convinced, but I guess we can't rely on that.

**Comment by sTiKyt** [comment]
> > ha How do you know if someone is a vegan: they WILL tell you! more proof of that right here.
> 
> Not being vegan means participating in violence against animals. In the same way, I can tell who is a speciesist who enjoys the suffering of animals.

Irrelevant, you're fighting a lost war. I'll eat animals because meat is tasty, good luck stopping me or people like me.. Oh, and the suffering? Well, people suffer just as much or even more, it's not a measure..

**Comment by sTiKyt** [comment]
> Let's try to be a bit more practical: I realize this application is written in C#, which is a Microsoft-associated and Windows-associated language AFAIK. But - I also know that there is some level of support for running C# on Linux. Microsoft claims it can be installed:
> 
> https://learn.microsoft.com/en-us/dotnet/core/install/linux
> 
> and maybe there are other alternatives. I've never used C# on Linux personally, so I'm just "spitballing" here.
> 
> Another alternative is porting the application. cloc says there are ~27K lines of C# code. Since this is a client for a network service, not a super-hyper-duper-special C# project, I would assume a reasonable developer who knows both C# and a reasonably-multiplatform language could do it. Perhaps even the developers of the app itself could be convinced, but I guess we can't rely on that.

C# is multiplatform, but only if you write the code properly and make sure it's not stuck depending on windows, so it's totally not a C# problem that they made it not work

**Comment by ThatOneCalculator** [comment]
Everyone, please take discussions of lifestyle choices somewhere else. Sorry to be blunt, but nobody wants to be on an email chain regarding veganism when this issue is supposed to be for desktop Linux support.
